### ISSUE TITLE
remove fgetln usage

### ISSUE THREAD

**Issue by bsdsx** [OpenSMTPD/OpenSMTPD#755]
Only 2 files use fgetln:

- contrib/libexec/encrypt/encrypt.c
- contrib/libexec/mail.local/mail.local.c

Replace fgetln with getline

**Comment by nickago** [comment]
I'm curious what the motivation behind this change is. Replacing `fgetln(3)` with `getline(3)` means losing reading variable length input. Thus, we will have to include arbitrary length constraints on the input, which I don't see as an improvement. What other benefits do we get from this change?

**Comment by poolpOrg** [comment]
OpenSMTPD used to rely on `fgetln(3)` until recently when all uses were replaced with `getline(3)` and these two files were not adapted only because they were only available in the portable version and outside the smtpd/ directory so they didn't get the same work. The benefit from this change is to bring the contrib/ up to date with the code constructs we already use in the daemon itself.

Now why did we change from `fgetln(3)` to `getline(3)` ?

Multiple reasons, but the primary one is that `getline(3)` is more portable than `fgetln(3)` which allows us to not rely on the portable version we used (and which turned out to be bugged at least once leading to issues on some systems), not to mention that the code construct to read input by lines when using `getline(3)` is simpler and less error-prone than the same code construct using `fgetln(3)`.

I don't understand the point regarding variable length input as I don't see a fundamental difference between the handling of length (_in our case_) but as far as OpenSMTPD is concerned in every place where we rely on a "read input line by line in a loop" contexts, we have arbitrary length constraints in upper levels anyways.
